### PR TITLE
Moving payments migrations models factories to package

### PR DESCRIPTION
**Pull request by phuclh**

### CODE REVIEW

File: config/payments.php [hunk 1]
```diff
@@ -2,4 +2,29 @@
 
 return [
 
+    /**
+     * User model
+     */
+    'user_model'     => \App\Models\User::class,
```
Comment (sl0wik) [inline config/payments.php:8]: Require update accordingly to https://www.notion.so/tger/Rules-c1a7ceb5485048d0b2b68ed4247f887b

Comment (phuclh) [reply config/payments.php]: Done!

File: database/migrations/2020_05_05_100000_create_payments_table.php.stub [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+use Illuminate\Support\Facades\Schema;
+
+class CreatePaymentsTable extends Migration
+{
+    public function up()
+    {
+        Schema::create('payments', function (Blueprint $table) {
+            $table->id();
+            $table->foreignId('order_id')->index()->references('id')->on('orders');
+            $table->foreignId('customer_id')->references('id')->on('customers'); // Defaults to customer_id on order, but allows payment to be made by different customer when paying an invoice
```
[the remark this answers is not in the document]
Comment (phuclh) [reply database/migrations/2020_05_05_100000_create_payments_table.php.stub]: Done!

**Review by joshtorres — APPROVED** [review]
Looks good. Thanks for making those changes

**Review by drewroberts — APPROVED** [review]
Thank you! 😷

**Review by joshtorres — COMMENTED** [review]
looks good

File: src/Models/Payment.php [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    protected $guarded = ['id'];
+    protected $casts = [
+    ];
+
+    protected static function boot()
+    {
+        parent::boot();
+
+        static::creating(function ($payment) {
+            if (empty($payment->creator_id) && auth()->check()) {
```
[the remark this answers is not in the document]
Comment (drewroberts) [reply src/Models/Payment.php:21]: Yes, I'll have someone go through and update the packages once that trait is added.

**Pull request merged**